src/huggingface_hub/utils/_errors.py

```diff
@@ -46,48 +51,60 @@ class EntryNotFoundError(HTTPError):
-        if error_code == "RepoNotFound":
-            raise RepositoryNotFoundError(
-                f"404 Client Error: Repository Not Found for url: {request.url}. If the"
-                " repo is private, make sure you are authenticated. (Request ID:"
-                f" {request_id})"
-            )
-        elif error_code == "RevisionNotFound":
-            raise RevisionNotFoundError(
-                f"404 Client Error: Revision Not Found for url: {request.url}. (Request"
-                f" ID: {request_id})"
-            )
-        elif error_code == "EntryNotFound":
-            raise EntryNotFoundError(
-                f"404 Client Error: Entry Not Found for url: {request.url}. (Request"
-                f" ID: {request_id})"
+    request_id = response.headers.get("X-Request-Id")
+    try:
+        response.raise_for_status()
+    except HTTPError as e:
+        if "X-Error-Code" in response.headers:
+            error_code = response.headers["X-Error-Code"]
+            if error_code == "RepoNotFound":
+                message = (
+                    f"{response.status_code} Client Error: Repository Not Found for"
+                    f" url: {response.url}. If the repo is private, make sure you are"
+                    f" authenticated. (Request ID: {request_id})"
+                )
+                raise RepositoryNotFoundError(message, response)
+            elif error_code == "RevisionNotFound":
+                message = (
+                    f"{response.status_code} Client Error: Revision Not Found for url:"
+                    f" {response.url}. (Request ID: {request_id})"
+                )
+                raise RevisionNotFoundError(message, response)
+            elif error_code == "EntryNotFound":
+                message = (
+                    f"{response.status_code} Client Error: Entry Not Found for url:"
+                    f" {response.url}. (Request ID: {request_id})"
+                )
+                raise EntryNotFoundError(message, response)
+
+        if response.status_code == 401:
+            # The repo was not found and the user is not Authenticated
+            message = (
+                f"{response.status_code} Client Error: Repository Not Found for url:"
+                f" {response.url}. If the repo is private, make sure you are"
+                f" authenticated. (Request ID: {request_id})"
             )
+            raise RepositoryNotFoundError(message, response)
 
-    if request.status_code == 401:
-        # The repo was not found and the user is not Authenticated
-        raise RepositoryNotFoundError(
-            f"401 Client Error: Repository Not Found for url: {request.url}. If the"
-            " repo is private, make sure you are authenticated. (Request ID:"
-            f" {request_id})"
-        )
+        if request_id is not None and len(e.args) > 0 and isinstance(e.args[0], str):
+            e.args = (e.args[0] + f" (Request ID: {request_id})",) + e.args[1:]
```

Comment:
move this to beginning of function to always do it, no?

Reply:
It was not needed before since it was already part of the error messages (this just adds the (Request ID: 123) to the end of the error message. I did some refactoring though to not have to do that.